pipeline-model-api/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/ast/ModelASTStage.java

```diff
@@ -26,18 +27,34 @@
     private ModelASTEnvironment environment;
     private Boolean failFast;
     private ModelASTStages parallel;
+    private List<ModelASTParallelContent> parallelContent = new ArrayList<>();
 
     public ModelASTStage(Object sourceLocation) {
         super(sourceLocation);
     }
 
+    protected Object readResolve() throws IOException {
+        // If there's already a set of parallel stages defined, add that to the parallel content instead.
+        if (this.parallel != null) {
+            this.parallelContent.addAll(this.parallel.getStages());
+            this.parallel = null;
```

Comment:
So `parallel` should now be `transient` and `@Deprecated`?

Reply:
Good point. `getParallel()` is deprecated already, and I've re-added `setParallel(ModelASTStages)` solely so I can deprecate it. =) Adding `transient` and `@Deprecated` on the field too.

Reply:
It is very strange to add a method just to deprecate it :/